core/src/main/java/com/scalar/db/transaction/consensuscommit/ParallelExecutor.java

```diff
@@ -113,9 +123,27 @@ public void rollbackRecords(List<ParallelExecutorTask> tasks, String transaction
           false,
           "rollbackRecords",
           transactionId);
-    } catch (ValidationConflictException e) {
+    } catch (ValidationConflictException | CrudException e) {
+      throw new AssertionError(
+          "Tasks for rolling back a transaction should not throw ValidationConflictException and CrudException",
+          e);
+    }
+  }
+
+  public void fillReadSetForRecordsFromWriteAndDeleteSetsIfUnread(
+      List<ParallelExecutorTask> tasks, String transactionId) throws CrudException {
+    try {
+      executeTasks(
+          tasks,
+          config.isParallelFillReadSetEnabled(),
+          false,
+          true,
+          "fillReadSetForRecordsFromWriteAndDeleteSetsIfUnread",
+          transactionId);
+    } catch (ExecutionException | ValidationConflictException e) {
```

Comment:
[minor] The order of exception classes is opposite to the above existing methods

Reply:
Let me fix this. Thanks.

Reply:
Ah, the order is consistently `ExecutionException`, `ValidationConflictException`, and `CrudException`. So I don't think we need to fix it. What do you think?

Reply:
Oops. I confused `ExecutionException` with `CrudException`. You're right. Please ignore my comment above.